traitlets/traitlets.py

```diff
@@ -407,11 +409,19 @@ def _dynamic_default_callable(self, obj):
         If neither exist, it returns None
         """
         # Traitlets without a name are not on the instance, e.g. in List or Union
+
         if self.name:
             mro = type(obj).mro()
-            meth_name = '_%s_default' % self.name
+            for cls in mro:
```

Comment:
@SylvainCorlay if I remember correctly, this logic here involving the class mro (which answers to `test_traitlets.test_dynamic_initializer`) is why installing default handler information on instances won't work. It could be stored on individual classes, however it can't be shared with subclasses, which means it would have to be installed using the metaclass. Not surprisingly that causes a lot of undesirable behavior when trying to use mixins (and probably other things as well).